Shell add rtt backend sample

jakub-uC opened the pull request:
Added RTT console to two Nordic boards: PCA10040 and PCA10056.
Added CI tests for Shell's RTT backend.

Commit:
boards: add rtt console to nrf52 boards

Added RTT Console to following boards:
- nrf52_pca10040
- nrf52840_pca10056

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

anangl commented on samples/subsys/shell/shell_module/sample.yaml:
```diff
@@ -2,13 +2,18 @@ sample:
 tests:
   test:
-    filter: ( CONFIG_UART_CONSOLE and CONFIG_SERIAL_SUPPORT_INTERRUPT )
+    filter: ( CONFIG_SERIAL)
```
What is the motivation of this change (and similar one a few lines below)? "Adding shell_rtt.c to sanity check" does not seem to explain such need.

jarz-nordic replied:
These were old shell requirements for old shell subsusystem. I can add this change separate commit.
New shell requires only  serial to be running, legacy one was depending on UART_CONSOLE and RX interrupt.

Current shell does not use UART_CONSOLE at all and can work with or without interrupts support.

anangl replied:
> I can add this change in separate commit.

I think this would be good, to avoid potential confusion.

jarz-nordic replied:
fixed

Commit:
shell: add RTT backend to CI tests

Adding shell_rtt.c to sanity check - related to issue #12860.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

Commit:
shell: update shell_uart backend test condition

New shell does not need either CONFIG_UART_CONSOLE
or interrupts support to be working. Only CONFIG_SERIAL
in combination with CONFIG_UART_SHELL_ON_DEV_NAME are required.
Updated sample yaml file accordingly.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>
